Feature/vizbox

LoyVanBeek opened the pull request.

LoyVanBeek committed:
Start adding launch file for RoboCup@Home Vizbox

LoyVanBeek committed:
Listen to the HMI for what the user says

LoyVanBeek committed:
Remove 'amigo' from topics, for use in a namespaced launch file

LoyVanBeek committed:
Include image from robot's head in vizbox

LoyVanBeek committed:
Move vizbox to interaction launch files

LoyVanBeek committed:
Output to log instead of screen

The vizbox is not a ROS node per se, so no default use of ROS logging either

LoyVanBeek committed:
Specify robot_name to supply to tools_tools transform because it doesn't handle namespaces

MatthijsBurgh commented on launch/interaction/vizbox.launch:
```diff
@@ -6,13 +6,15 @@
 
     <node pkg="topic_tools" type="transform" name="extract_hmi_result_sentence"
-        args="hmi/result hmi/result/sentence std_msgs/String 'm.result.sentence'"/>
+        args="/$(arg robot_name)/hmi/result hmi/result/sentence std_msgs/String 'm.result.sentence'"/>
```
Or include this launch file in the robot ns?

LoyVanBeek replied:
It is, but topic_tools doesn't handle namespaces AFAICT. 
hmi/result is in the same namepace and the ROS namespacing works for that.

LoyVanBeek replied:
Of course I tried without ```/$(arg robot_name)/```, but that results in an error:

```
ERROR: Wrong input topic (or topic field): hmi/result
[amigo/extract_hmi_result_sentence-18] process has died [pid 12326, exit code 1, cmd /opt/ros/kinetic/lib/topic_tools/transform hmi/result hmi/result/sentence std_msgs/String m.result.sentence __name:=extract_hmi_result_sentence __log:=/home/loy/.ros/log/7310f27a-4d6e-11e8-bcb2-d07e35c47b06/amigo-extract_hmi_result_sentence-18.log].
log file: /home/loy/.ros/log/7310f27a-4d6e-11e8-bcb2-d07e35c47b06/amigo-extract_hmi_result_sentence-18*.log
```

MatthijsBurgh replied:
👍

MatthijsBurgh replied:
But does it output to `/robot_name/hmi/result/sentence`? Or to `hmi/result/sentence`? What is the desired behaviour? It is without the robot name. Please make it absolute.

jlunenburg commented on launch/interaction/vizbox.launch:
```diff
@@ -0,0 +1,20 @@
+<launch>
+
+    <!-- Get the machine file -->
+    <arg name="machine" default="localhost"/>
```
I guess machine files aren't used anymore?!

LoyVanBeek replied:
Yes they are, all the state machine launches declare which machine to run each node in. I just forgot to pass that on to the actual node

jlunenburg commented on launch/interaction/vizbox.launch:
```diff
@@ -0,0 +1,20 @@
+    <arg name="robot_name"/>
+
+    <node pkg="vizbox" type="server.py" name="vizbox" cwd="node" output="log">
+        <remap to="text_to_speech/output" from="robot_text"/>
```
Being picky, isn't it easier to read from .. to?

jlunenburg commented on launch/interaction/vizbox.launch:
```diff
@@ -0,0 +1,20 @@
+
+    <arg name="robot_name"/>
+
+    <node pkg="vizbox" type="server.py" name="vizbox" cwd="node" output="log">
```
Is vizbox installed by default?

LoyVanBeek replied:
https://github.com/tue-robotics/tue-env/pull/261

jlunenburg reviewed (changes requested):
Have a look at the comments. If you don't use a machine files, don't include them? Furthermore, did you test this on the robot?

LoyVanBeek committed:
Start server on specified machine

LoyVanBeek committed:
Remap in order 'from' and then 'to'

Should be easier to read. From means 'the node calls it this internally'; to means 'this is what the rest of the system should use'

MatthijsBurgh commented:
@LoyVanBeek please decide what to with the vizbox.

LoyVanBeek commented:
I think it's handy for us to have regards of RoboCup, so I'd say merge it ASAP

LoyVanBeek commented:
Nobody cares

LoyVanBeek committed:
Resolved some review comments

LoyVanBeek committed:
Throttle parametrized image stream

MatthijsBurgh commented on launch/interaction/vizbox.launch:
```diff
@@ -0,0 +1,24 @@
+    <!-- Get the machine file -->
+    <arg name="machine" default="localhost"/>
+    <arg name="image" default="top_kinect/rgb/image"/>
+    <include file="$(env ROBOT_BRINGUP_PATH)/machines/$(arg machine).machine" />
```
better to put this `include` directly after the machine argument.

LoyVanBeek committed:
Reorder some lines to group them
